### PR TITLE
Don't copy authorization header during redirects in MvcTesting

### LINKED ISSUE #20896: Testing: RedirectHandler incorrectly copies Authorization header to redirected requests
### Describe the bug
The [RedirectHandler](https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs) copies the `Authorization` header to redirected requests which is inconsistent behavior compared to the [default behavior](https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.allowautoredirect?redirectedfrom=MSDN&view=netcore-3.1#remarks) of HttpClient.

This will incorrectly make tests pass that will fail when the tested routes are connected to with a real HttpClient.

### To Reproduce

The following test does not pass:

```
public class RedirectTests : IClassFixture<WebApplicationFactory>
    {
        private readonly HttpClient _client;

        public DocumentsTests(CustomWebApplicationFactory factory)
        {
            _client = factory.CreateClient();
            _client.DefaultRequestHeaders.Add(HeaderNames.Authorization, $"Bearer key");
        }

        [Fact]
        public async Task ShouldReturnUnautorhizedWhenRedirected()
        {
            var response = await _client.PostAsync("/EndpointThatReturnsRedirectToARouteThatRequiresAuthentication", content);
            Assert.Equal(401, response.StatusCode);
        }
   }
```

### Further technical details
- ASP.NET Core version 3.1
- VS

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/dotnet/aspnetcore/issues/20896.

### CODE REVIEW

**Comment by captainsafia** [comment]
@pranavkm Ping for review.

File: src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs [hunk 1]
```diff
@@ -104,7 +105,13 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
     {
         foreach (var header in originalRequestHeaders)
         {
-            newRequestHeaders.TryAddWithoutValidation(header.Key, header.Value);
+            // Avoid copying the Authorization header to match the behavior
+            // in the built-in HTTP client when processing redirects
+            // https://github.com/dotnet/runtime/blob/69b5d67d9418d672609aa6e2c418a3d4ae00ad18/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L509-L517
+            if (header.Key != HeaderNames.Authorization)
+            {
+                newRequestHeaders.TryAddWithoutValidation(header.Key, header.Value);
```
Comment (BrennanConroy) [inline src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs:113]: nit: could the param be renamed to `redirectRequestHeaders`

**Review by BrennanConroy — APPROVED** [review]

File: src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs [hunk 2]
```diff
@@ -104,7 +105,13 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
     {
         foreach (var header in originalRequestHeaders)
         {
-            newRequestHeaders.TryAddWithoutValidation(header.Key, header.Value);
+            // Avoid copying the Authorization header to match the behavior
+            // in the built-in HTTP client when processing redirects
+            // https://github.com/dotnet/runtime/blob/69b5d67d9418d672609aa6e2c418a3d4ae00ad18/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L509-L517
+            if (header.Key != HeaderNames.Authorization)
```
Comment (pranavkm) [inline src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs:111]: Does this need a `OrdinalIgnoreCase` comparison? Guessing that's a string on the LHS, no?

File: src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs [hunk 3]
```diff
@@ -104,7 +105,13 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
     {
         foreach (var header in originalRequestHeaders)
         {
-            newRequestHeaders.TryAddWithoutValidation(header.Key, header.Value);
+            // Avoid copying the Authorization header to match the behavior
+            // in the built-in HTTP client when processing redirects
```
Comment (pranavkm) [inline src/Mvc/Mvc.Testing/src/Handlers/RedirectHandler.cs:109]:
```suggestion
            // HttpClient when processing redirects
```
?

**Review by pranavkm — APPROVED** [review]

**Pull request merged**